src/cloudFlowUtils/models.ts

```diff
@@ -17,23 +18,28 @@ export interface SubnetGroup {
 export const GroupType: Record<string, GroupType> = {
   Region: {
     id: "Region",
-    getGroupId: (subnet: Subnet) => `REGION_${subnet.region}`,
+    getGroupId: (subnet: Subnet) => (subnet ? `REGION_${subnet.region}` : ""),
   },
   Vpc: {
     id: "Vpc",
-    getGroupId: (subnet: Subnet) => `VPC_${subnet.vpc}`,
+    getGroupId: (subnet: Subnet) => (subnet ? `VPC_${subnet.vpc}` : ""),
   },
   Subnet: {
     id: "Subnet",
-    getGroupId: (subnet: Subnet) => `SUBNET_${subnet.subnet}`,
+    getGroupId: (subnet: Subnet) => (subnet ? `SUBNET_${subnet.subnet}` : ""),
```

Comment:
Is it expected that subnet is null in some cases? If so I think we should avoid calling this function altogether with an external check for that. But I need to look more closely at the reasoning here to understand the change better.

If it is expected to be null/undefined, we should also capture that in the TypeScript typings... e.g. `subnet: Subnet | null`. Then the compiler will complain that we didn't handle it. But, still think we should avoid calling this in that case

Reply:
The subnet might be null/undefined only on the links elements after my changes with the raw data handling.
I will adopt your suggestion to not call the function when these cases occur.